cmd/thanos/store.go

```diff
@@ -129,6 +130,9 @@ func (sc *storeConfig) registerFlag(cmd extkingpin.FlagClause) {
 	cmd.Flag("block-meta-fetch-concurrency", "Number of goroutines to use when fetching block metadata from object storage.").
 		Default("32").IntVar(&sc.blockMetaFetchConcurrency)
 
+	cmd.Flag("debug.series-batch-size", "The batch size when fetching series from object storage.").
```

Comment:
Also I am not sure this is usable - if you have 2000 blocks you have 10000  * 2000 batch size kind of, and when store has one block it has 10000... Should we have global to request batch size? (otherwise name and help should change)

Reply:
The batch size is per block, and each block is retrieved concurrently through a separate goroutine. We can also try to infer the batch size based on the number of total expanded postings.